File: captain/types/flowchart.py
# Copyright (c) 2024 Emasoft
#
# This file is part of AtlasVibe, which is based on Flojoy Studio
# Original Copyright (c) 2023-2024 Flojoy
#
# SPDX-License-Identifier: MIT
# See the LICENSE file for details.

from pydantic import BaseModel


class PostCancelFC(BaseModel):
    jobsetId: str | None = None


class PostWFC(BaseModel):
    fc: str
    jobsetId: str
    cancelExistingJobs: bool
    observeBlocks: list[str]
    nodeDelay: float
    maximumRuntime: float
    maximumConcurrentWorkers: int
    projectPath: str | None = None


class WorkerSuccessResponse(BaseModel):
    node_id: str
    result: dict


class WorkerFailedResponse(BaseModel):
    node_id: str
    result: dict

File: src/renderer/stores/flowchart.ts
/*
 * Copyright (c) 2024 Emasoft
 *
 * This file is part of AtlasVibe, which is based on Flojoy Studio
 * Original Copyright (c) 2023-2024 Flojoy
 *
 * SPDX-License-Identifier: MIT
 * See the LICENSE file for details.
 */

import { XYPosition } from "reactflow";
import { create } from "zustand";
import { immer } from "zustand/middleware/immer";

type State = {
  isEditMode: boolean;
  centerPosition: XYPosition;
  nodeParamChanged: boolean;
};

type Actions = {
  setIsEditMode: (val: boolean) => void;
  setCenterPosition: (position: XYPosition) => void;
  markNodeParamChanged: () => void;
  resetNodeParamChanged: () => void;
};

export const useFlowchartStore = create<State & Actions>()(
  immer((set) => ({
    isEditMode: false,
    setIsEditMode: (val) =>
      set((state) => {
        state.isEditMode = val;
      }),

    centerPosition: { x: 0, y: 0 },
    setCenterPosition: (position) =>
      set((state) => {
        state.centerPosition = position;
      }),

    nodeParamChanged: false,
    markNodeParamChanged: () => set({ nodeParamChanged: true }),
    resetNodeParamChanged: () => set({ nodeParamChanged: false }),
  })),
);
